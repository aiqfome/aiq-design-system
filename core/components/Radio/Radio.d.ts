import React, { InputHTMLAttributes } from 'react';
import { MarginProps } from 'styled-system';
export interface Props extends MarginProps, InputHTMLAttributes<HTMLInputElement> {
    name: string;
    value: any;
    variant?: 'default' | 'small';
    disabled?: boolean;
    label?: string;
    defaultChecked?: boolean;
    onChange?: (event: any) => void;
}
export declare const Radio: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLInputElement>>;
//# sourceMappingURL=Radio.d.ts.map