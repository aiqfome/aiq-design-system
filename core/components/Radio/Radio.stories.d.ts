import React from 'react';
declare const _default: {
    component: React.ForwardRefExoticComponent<import("./Radio").Props & React.RefAttributes<HTMLInputElement>>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const Basic: React.FC;
export declare const Disabled: React.FC;
export declare const Checked: React.FC;
export declare const CheckedDisabled: React.FC;
export declare const Label: React.FC;
export declare const Sizes: React.FC;
//# sourceMappingURL=Radio.stories.d.ts.map