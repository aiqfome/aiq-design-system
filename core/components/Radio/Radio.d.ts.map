{"version":3,"file":"Radio.d.ts","sourceRoot":"","sources":["../../../src/components/Radio/Radio.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,mBAAmB,EAAE,MAAM,OAAO,CAAA;AAIlD,OAAO,EAAU,WAAW,EAAE,MAAM,eAAe,CAAA;AAKnD,MAAM,WAAW,KACf,SAAQ,WAAW,EACjB,mBAAmB,CAAC,gBAAgB,CAAC;IACvC,IAAI,EAAE,MAAM,CAAA;IACZ,KAAK,EAAE,GAAG,CAAA;IACV,OAAO,CAAC,EAAE,SAAS,GAAG,OAAO,CAAA;IAC7B,QAAQ,CAAC,EAAE,OAAO,CAAA;IAClB,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,OAAO,CAAC,EAAE,OAAO,CAAA;IACjB,QAAQ,CAAC,EAAE,CAAC,KAAK,EAAE,GAAG,KAAK,IAAI,CAAA;CAChC;AA2FD,eAAO,MAAM,KAAK,gFAiDjB,CAAA"}