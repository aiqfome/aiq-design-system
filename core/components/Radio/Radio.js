"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Radio = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const Text_1 = require("../Text");
const Box_1 = require("../Box");
const radioVariations = {
    default: styled_components_1.css `
    min-height: 21px;
    padding-left: 35px;
    div {
      height: 20px;
      width: 20px;
    }

    div:after {
      top: 3px;
      left: 3px;
      width: 10px;
      height: 10px;
    }
  `,
    small: styled_components_1.css `
    min-height: 17px;
    padding-left: 28px;
    div {
      height: 16px;
      width: 16px;
    }

    div:after {
      top: 2px;
      left: 2px;
      width: 8px;
      height: 8px;
    }
  `
};
const RadioStyled = styled_components_1.default.label `
  ${styled_system_1.margin}

  &:hover {
    cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  }
  display: flex;
  align-items: center;
  position: relative;
  cursor: pointer;

  opacity: ${props => (props.disabled ? 0.5 : 1)};

  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  div {
    position: absolute;
    top: 0;
    left: 0;
    border-radius: 50%;
    border: 2px solid ${({ theme }) => theme.colors.primary};
  }

  input:checked ~ div {
    background-color: #fff;
  }

  div:after {
    content: '';
    position: absolute;
    display: none;
  }

  input:checked ~ div:after {
    display: block;
  }

  div:after {
    border-radius: 50%;
    background-color: ${({ theme }) => theme.colors.primary};
  }

  ${({ variant }) => radioVariations[variant || 'default']}
`;
exports.Radio = react_1.default.forwardRef((_a, ref) => {
    var { name, value, disabled = false, checked = false, label, variant = 'default', onChange = () => {
        // do nothing.
    }, className, mx, my, m, mr, ml } = _a, props = __rest(_a, ["name", "value", "disabled", "checked", "label", "variant", "onChange", "className", "mx", "my", "m", "mr", "ml"]);
    const [isChecked, setIsChecked] = react_1.useState(checked);
    react_1.useEffect(() => {
        setIsChecked(checked);
    }, [checked]);
    const onChangeRadio = e => {
        if (!disabled) {
            setIsChecked(!checked);
            if (onChange)
                onChange(e);
        }
    };
    return (react_1.default.createElement(RadioStyled, { mx: mx, my: my, variant: variant, m: m, mr: mr, ml: ml, className: className, disabled: disabled, "data-testid": 'radio-container' },
        react_1.default.createElement("input", Object.assign({ ref: ref, type: 'radio', disabled: disabled, name: name, onChange: onChangeRadio, checked: isChecked, value: value, "data-testid": 'radio-input' }, props)),
        react_1.default.createElement(Box_1.Box, null),
        label && react_1.default.createElement(Text_1.Text, null, label)));
});
exports.Radio.displayName = 'Radio';
exports.Radio.propTypes = {
    name: prop_types_1.default.string.isRequired,
    value: prop_types_1.default.any.isRequired,
    disabled: prop_types_1.default.bool,
    checked: prop_types_1.default.bool,
    onChange: prop_types_1.default.func,
    label: prop_types_1.default.string,
    className: prop_types_1.default.string,
    variant: prop_types_1.default.oneOf(['small', 'default']),
    mx: prop_types_1.default.any,
    my: prop_types_1.default.any,
    m: prop_types_1.default.any,
    mr: prop_types_1.default.any,
    ml: prop_types_1.default.any
};
