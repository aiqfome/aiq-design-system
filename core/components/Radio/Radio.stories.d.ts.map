{"version":3,"file":"Radio.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Radio/Radio.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;;AAMzB,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAWzB,CAAA;AAED,eAAO,MAAM,QAAQ,EAAE,KAAK,CAAC,EAI5B,CAAA;AAED,eAAO,MAAM,OAAO,EAAE,KAAK,CAAC,EAI3B,CAAA;AAED,eAAO,MAAM,eAAe,EAAE,KAAK,CAAC,EAUnC,CAAA;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAKzB,CAAA;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAmBzB,CAAA"}