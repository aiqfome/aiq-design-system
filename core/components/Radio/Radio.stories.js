"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sizes = exports.Label = exports.CheckedDisabled = exports.Checked = exports.Disabled = exports.Basic = void 0;
const react_1 = __importDefault(require("react"));
const addon_knobs_1 = require("@storybook/addon-knobs");
const Radio_1 = require("./Radio");
const Flex_1 = require("../Flex");
exports.default = {
    component: Radio_1.Radio,
    title: 'Radio',
    decorators: [addon_knobs_1.withKnobs]
};
exports.Basic = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { disabled: addon_knobs_1.boolean('disabled', false), checked: addon_knobs_1.boolean('checked', true), mx: 10, name: 'radioBasic', value: '01' }),
    react_1.default.createElement(Radio_1.Radio, { name: 'radioBasic', value: '02' })));
exports.Disabled = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { mx: 10, disabled: true, name: 'radioDisabled', value: '01' })));
exports.Checked = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { mx: 10, checked: true, name: 'radioChecked', value: '01' })));
exports.CheckedDisabled = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { mx: 10, disabled: true, checked: true, name: 'radioCheckedDisabled', value: '01' })));
exports.Label = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { mx: 10, checked: true, label: 'aiq', name: 'radio', value: 'aiq' }),
    react_1.default.createElement(Radio_1.Radio, { mx: 10, checked: false, label: 'fome', name: 'radio', value: 'fome' })));
exports.Sizes = () => (react_1.default.createElement(Flex_1.Flex, null,
    react_1.default.createElement(Radio_1.Radio, { mx: 10, variant: 'default', label: 'default', checked: true, name: 'radio1', value: 'default' }),
    react_1.default.createElement(Radio_1.Radio, { mx: 10, checked: true, variant: 'small', label: 'small', name: 'radio2', value: 'small' })));
