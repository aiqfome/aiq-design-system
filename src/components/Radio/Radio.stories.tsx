import React from 'react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import { Radio } from './Radio'
import { Flex } from '../Flex'

export default {
  component: Radio,
  title: 'Radio',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => (
  <Flex>
    <Radio
      disabled={boolean('disabled', false)}
      checked={boolean('checked', true)}
      mx={10}
      name='radioBasic'
      value='01'
    />
    <Radio name='radioBasic' value='02' />
  </Flex>
)

export const Disabled: React.FC = () => (
  <Flex>
    <Radio mx={10} disabled={true} name='radioDisabled' value='01' />
  </Flex>
)

export const Checked: React.FC = () => (
  <Flex>
    <Radio mx={10} checked={true} name='radioChecked' value='01' />
  </Flex>
)

export const CheckedDisabled: React.FC = () => (
  <Flex>
    <Radio
      mx={10}
      disabled={true}
      checked={true}
      name='radioCheckedDisabled'
      value='01'
    />
  </Flex>
)
