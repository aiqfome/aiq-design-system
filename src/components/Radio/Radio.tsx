import React, { InputHTMLAttributes } from 'react'

import PropTypes from 'prop-types'

import styled, { DefaultTheme } from 'styled-components'

import { margin, MarginProps } from 'styled-system'

import { Text } from '../Text'
import { Box } from '../Box'

export interface Props
  extends MarginProps,
    InputHTMLAttributes<HTMLInputElement> {
  name: string
  value: any
  disabled?: boolean
  label?: string
  checked?: boolean
  onChange?: (event: any) => void
}

interface RadioStyled extends DefaultTheme, MarginProps {
  disabled: boolean
}

const RadioStyled = styled.label<RadioStyled>`
  ${margin}

  &:hover {
    cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  }
  min-height: 21px;
  display: flex;
  align-items: center;
  position: relative;
  padding-left: 35px;
  cursor: pointer;

  opacity: ${props => (props.disabled ? 0.5 : 1)};

  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  div {
    position: absolute;
    top: 0;
    left: 0;
    height: 20px;
    width: 20px;
    border-radius: 50%;
    border: 2px solid ${({ theme }) => theme.colors.primary};
  }

  input:checked ~ div {
    background-color: #fff;
  }

  div:after {
    content: '';
    position: absolute;
    display: none;
  }

  input:checked ~ div:after {
    display: block;
  }

  div:after {
    top: 3px;
    left: 3px;
    width: 10px;
    height: 10px;
    border-radius: 50%;
    background-color: ${({ theme }) => theme.colors.primary};
  }
`

export const Radio: React.FC<Props> = ({
  name,
  value,
  disabled = false,
  checked = false,
  label,
  onChange = () => {
    // do nothing.
  },
  className,
  mx,
  my,
  m,
  mr,
  ml,
  ...props
}) => {
  return (
    <RadioStyled
      mx={mx}
      my={my}
      m={m}
      mr={mr}
      ml={ml}
      className={className}
      disabled={disabled}
    >
      <input
        type='radio'
        disabled={disabled}
        name={name}
        onChange={onChange}
        checked={checked}
        value={value}
        {...props}
      />
      <Box />

      {label && <Text>{label}</Text>}
    </RadioStyled>
  )
}

Radio.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.any.isRequired,
  disabled: PropTypes.bool,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  label: PropTypes.string,
  className: PropTypes.string,
  mx: PropTypes.any,
  my: PropTypes.any,
  m: PropTypes.any,
  mr: PropTypes.any,
  ml: PropTypes.any
}
