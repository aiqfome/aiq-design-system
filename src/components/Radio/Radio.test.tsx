import React from 'react'

import { Radio } from './Radio'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Radio name='radio' value={true} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
