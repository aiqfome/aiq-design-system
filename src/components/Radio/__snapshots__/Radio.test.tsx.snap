// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<label
  className="sc-AxiKw ibRvfq"
  disabled={false}
>
  <input
    checked={false}
    disabled={false}
    name="radio"
    onChange={[Function]}
    type="radio"
    value={true}
  />
  <div
    className="sc-AxirZ RoPcv"
  />
</label>
`;
